tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1103,7 +1131,22 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.insert(operands.begin() + 2, MaybeConstantFoldBias(bias));
+    HloInstruction *maybe_constant_folded_bias = MaybeConstantFoldBias(bias);
+    // if (slice && bitcast) {
+    if (bitcast) {
+      maybe_constant_folded_bias =
+          instr->AddInstruction(HloInstruction::CreateBitcast(
+              slice->shape(), maybe_constant_folded_bias));
+    }
+
+    if (gemm->custom_call_target() == kCublasLtMatmulF8CallTarget ||
+        gemm->custom_call_target() == kCublasLtMatmulCallTarget ||
+        gemm->custom_call_target() == kGemmCallTarget) {
```

Comment:
I think `gemm` is guaranteed to be a cublas or cublas LT custom call, so no need to check this.

Reply:
Read from [here](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/xla/service/gpu/cublas_cudnn.cc;l=29?q=IsLegacyCublasMatmul&sq=&ss=tensorflow%2Ftensorflow) there are 3 kinds of cublas or cublaslt calls.